### ISSUE TITLE
Keep manifest up-to-date with running job

### ISSUE THREAD

**Issue by iszulcdeepsense** [TheRacetrack/racetrack#250]
According to https://github.com/TheRacetrack/racetrack/issues/217#issuecomment-1476204943, it can be surprising for users if we allow to have Job's manifest out of sync with the actual job.
It would be better to automatically redeploy it (rebuild and reprovision) to make sure it always reflects real state.

**Comment by LookACastle** [comment]
Making sure that that comment and in-meeting discussions are understood properly.

With #244, we introduce the ability to edit manifests online. This is a good change, but it leaves the manifest to be - for all intents and purposes - wrong until reprovisioned (and, sometimes rebuilt and reprovisioned). Wrong information is no good, the manifest should reflect what is currently running - not what would be running where someone to press the right set of buttons. As such, I recommended redeploying on online edit, yes. This is a good change that I'd welcome.

The comment linked is talking about a different way of being out of sync. Namely, that a developer would edit the manifest online, get their job working, and then go about being happy with their change. This would be undesirable, because that manifest exists on the racetrack. A manifest deployed from online, should in my opinion be marked "Out of sync" or similar regardless of what other circumstances exist, because it is not being tracked by the developers version control at all, and it'd be difficult to redeploy the job under this circumstance. This is nudging us a bit in the direction of gitops, where a manifest exists in a git repository rather than being tracked by the client.

Whether we should be proper gitops is out of scope for this discussion, but I do think we should take a page out of their book, and strongly nudge people towards not skipping version control. Manifests edited online are all but guaranteed to be improperly version controlled, so we should mark them as such.

TL;DR: Good change, I approve, but I feel the comment linked is not relevant to this.

**Comment by iszulcdeepsense** [comment]
@LookACastle Indeed, now I see what you really meant a linked comment.
